tools/sdk-testgen/packages/autorest.gotest/doc/gotest_howto.md

```diff
@@ -0,0 +1,156 @@
+- [Node 1.14+](https://nodejs.org/en/download/) is required and [Autorest CLI](https://github.com/Azure/autorest/tree/main/packages/apps/autorest) should be installed globally.
+
+## Config swagger
+
+We use `autorest.md` file under SDK module folder root (`<sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr/autorest.md`) to config the service swagger file to be used for code generation. Change the file content as follows.
+
+````md
+### AutoRest Configuration
+
+> see https://aka.ms/autorest
+
+```yaml
+azure-arm: true
+require:
+- <swagger-repo-workspace>/tools/sdk-testgen/swagger/specification/signalr/resource-manager/readme.md
+- <swagger-repo-workspace>/tools/sdk-testgen/swagger/specification/signalr/resource-manager/readme.go.md
+license-header: MICROSOFT_MIT_NO_VERSION
+module-version: 0.5.0
+```
+````
+
+
+## Generate SDK
+
+After you set the swagger config, you need to regenerate the SDK first. (You could use `--generate-sdk` to generate SDK along with example/test/sample generation to skip this step.)
+
+1. Delete all `.go` files under the module folder.
+```sh
+rm <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr/*.go
+```
+2. Use `@autorest/go` to generate SDK.
+```sh
+autorest --version=3.8.2 --use=@autorest/go@latest --go --track2 --output-folder=<sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr --file-prefix="zz_generated_" --clear-output-folder=false <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr/autorest.md
+```
+3. Resolve dependency.
+```sh
+cd <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr
+go mod tidy
+```
+
+## Generate examples
+
+We use [swagger defined examples](https://github.com/Azure/azure-rest-api-specs/blob/main/document/x-ms-examples.md) to generate example code for each operation. All the examples released with the SDK modules will be displayed in `pkg.go.dev` following the operation's API reference.
+
+1. Use `@autorest/gotest` to generate the example code. If you want to generate the SDK code as well, add `--generate-sdk` tag.
+```sh
+autorest --version=3.8.2 --use=@autorest/go@latest --use=@autorest/gotest@latest --go --track2 --output-folder=<sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr --clear-output-folder=false --go.clear-output-folder=false --testmodeler.generate-sdk-example <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr/autorest.md
+```
+2. Resolve dependency.
+```sh
+cd <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr
+go mod tidy
+```
+
+## Generate mock test and execute
+
+We use [swagger defined examples](https://github.com/Azure/azure-rest-api-specs/blob/main/document/x-ms-examples.md) to generate mock tests. These tests can be executed with [mock service host](https://github.com/Azure/azure-sdk-tools/tree/main/tools/mock-service-host) to ensure the correctness of SDK.

```

Comment:
Can we have a script to put everything together?  As I know, dotnet already has that.  @changlong-liu

Reply:
We have example generation in release tool as well as mock test generation and execution in pipeline script for Go.

Reply:
The dotnet script here (https://github.com/Azure/azure-sdk-for-net/blob/main/eng/scripts/Launch-MockServiceHost.ps1) is   mainly for launching mock-service-host (for sub step 3 and 5 in this article) and trust the certificates in windows (no need to do this in Golang).